static/components/TaskDetails.jsx

```diff
@@ -124,40 +136,124 @@ var TaskDetails = React.createClass({
       task: data
     });
   },
+
+  splitLogData(data){
+    var text       = data.chunks;
+    var objLength  = data.chunks.length;
+
+    for(var i = 0; i < objLength; i++){
+      if(data.chunks[i] != undefined){
+        this.setState({
```

Comment:
I see two potential problems here:
* react batches calls to `setState`, but it might be better to just call `setState` once after your data is finished mutating.
* `setState` is async, so when you reference `this.state.lines` in `updateBuildLog`, it may not be what you expect after calling `splitLogData`.